src/modules/endpoint/sagas/watchClearResourcesAndCache.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+
+import type { DataContainer } from '../DataContainer'
+import type { Saga } from 'redux-saga'
+import { takeLatest, call, put } from 'redux-saga/effects'
+import type { FetchEventActionType } from '../../app/StoreActionType'
+
+export function * clearResourcesAndCache (dataContainer: DataContainer, action: FetchEventActionType): Saga<void> {
+  console.debug('Clearing Resource Cache')
+  dataContainer.clearCaches()
+  yield call(dataContainer.clearOfflineStorage)
+  yield put({ type: 'FETCH_CITIES', params: { forceRefresh: true } })
+}
+
+export default function * (dataContainer: DataContainer): Saga<void> {
```

Comment:
I'm not sure whether this requires a test.

Pro:
* It could be used in production code (e.g. button in settings) and we maybe forget to test it

Con:
* It is not a core saga

Reply:
I think we should test it, because i am sure that we need this functionality in the future in production